src/externalUser.ts

```diff
@@ -57,25 +150,75 @@ export class ExternalUser {
-     * fethces user from blockchain
-     * checks if user is loggedin by verifying the keys
-     * delete the keys from storage if they are not verified
-     * returns the user object
+     * Signs a message with the given key manager and the key level
+     *
+     * @param message {any} - an object to sign
+     * @param keyManager {KeyManager} - the key manager to use to sign the message
+     * @param keyManagerLevel {KeyManagerLevel=BROWSER_LOCAL_STORAGE} - the level to use to sign the message
      */
-    // return Object.assign(this, {})
-    //   }
+    static async signMessage(
+        message: any,
+        keyManager: KeyManager,
+        keyManagerLevel: KeyManagerLevel = KeyManagerLevel.BROWSER_LOCAL_STORAGE,
+        recipient?: string
+    ): Promise<Message> {
+        const publicKey = await keyManager.getKey({
+            level: keyManagerLevel,
+        });
+
+        if (!publicKey) throw throwError('No Key Found for this level', SdkErrors.KeyNotFound);
+        const signer = createVCSigner(keyManager, keyManagerLevel).sign;
+
+        const jwk = await createJWK(publicKey);
+
+        const issuer = toDid(jwk);
+
+        return await Message.sign(message, { did: issuer, signer: signer as any, alg: 'ES256K-R' }, recipient);
+    }
 
     /**
+     * Receives the login request from Tonomy ID and verifies the login was successful
+     *
+     * @description should be called in the callback page
      *
-     * @param keymanager
-     * @throws error if user didn't login correctly
-     * @returns external user objects
+     * @returns {Promise<ExternalUser>} an external user object ready to use
      */
-    //   static verifyLogin(keymanager = JSsKeymanager: KeyManager, storage = jsStorage: Storage):  Promise<ExternalUser>  {
-    //     userApps.callBack(keymanager);
-    //     return Object.assign(this, {})
-    //   }
+    static async verifyLoginRequest(checkKeys = true, keyManager?: KeyManager): Promise<ExternalUser> {
+        const { requests, username, accountName } = UserApps.getLoginRequestParams();
+
+        const result = await UserApps.verifyRequests(requests);
+
+        const myKeyManager = keyManager || new JsKeyManager();
+        const loginRequest = result.find((r) => r.getPayload().origin === window.location.origin)?.getPayload();
+
+        if (!loginRequest) throwError('No login request found for this origin', SdkErrors.OriginMismatch);
+        if (
+            loginRequest.publicKey !==
+            (await myKeyManager.getKey({ level: KeyManagerLevel.BROWSER_LOCAL_STORAGE }))?.toString()
+        )
+            throwError('Key in request does not match', SdkErrors.KeyNotFound);
+
+        if (checkKeys) {
+            const keyExists = await UserApps.verifyKeyExistsForApp(accountName, myKeyManager);
+
+            if (!keyExists) throwError('Key not found', SdkErrors.KeyNotFound);
+        }
+
+        const externalUser = new ExternalUser(myKeyManager, jsStorageFactory);
+
```

Comment:
if the user cant inject the storage then there is no reason for us to inject it ourself this could have been in the  
class {
private storage = new JsStorage()
}

Reply:
done